src/components/views/dialogs/CreateGroupDialog.js

```diff
@@ -0,0 +1,196 @@
+            groupName: '',
+            groupId: '',
+            groupError: null,
+            creating: false,
+            createError: null,
+        };
+    },
+
+    _onGroupNameChange: function(e) {
+        this.setState({
+            groupName: e.target.value,
+        });
+    },
+
+    _onGroupIdChange: function(e) {
+        this.setState({
+            groupId: e.target.value,
+        });
+    },
+
+    _onGroupIdBlur: function(e) {
+        this._checkGroupId();
+    },
+
+    _checkGroupId: function(e) {
+        const parsedGroupId = this._parseGroupId(this.state.groupId);
+        let error = null;
+        if (parsedGroupId === null) {
+            error = _t(
+                "Group IDs must be of the form +localpart:%(domain)s",
+                {domain: MatrixClientPeg.get().getDomain()},
+            );
+        } else {
+            const domain = parsedGroupId[1];
+            if (domain !== MatrixClientPeg.get().getDomain()) {
+                error = _t(
+                    "It is currently only possible to create groups on your own home server: "+
+                    "use a group ID ending with %(domain)s",
+                    {domain: MatrixClientPeg.get().getDomain()},
+                );
+            }
+        }
+        this.setState({
+            groupIdError: error,
+        });
+        return error;
+    },
+
+    _onFormSubmit: function(e) {
+        e.preventDefault();
+
+        if (this._checkGroupId()) return;
+
+        const parsedGroupId = this._parseGroupId(this.state.groupId);
+        const profile = {};
+        if (this.state.groupName !== '') {
+            profile.name = this.state.groupName;
+        }
+        this.setState({creating: true});
+        MatrixClientPeg.get().createGroup({
+            localpart: parsedGroupId[0],
+            profile: profile,
+        }).then((result) => {
+            dis.dispatch({
+                action: 'view_group',
+                group_id: result.group_id,
+            });
+            this.props.onFinished(true);
+        }).catch((e) => {
+            this.setState({createError: e});
+        }).finally(() => {
+            this.setState({creating: false});
+        }).done();
+    },
+
+    _onCancel: function() {
+        this.props.onFinished(false);
+    },
+
+    /**
+     * Parse a string that may be a group ID
+     * If the string is a valid group ID, return a list of [localpart, domain],
+     * otherwise return null.
+     *
+     * @param {string} groupId The ID of the group
+     * @return {string[]} array of localpart, domain
+     */
+    _parseGroupId: function(groupId) {
+        const matches = GROUP_REGEX.exec(this.state.groupId);
+        if (!matches || matches.length < 3) {
+            return null;
+        }
+        return [matches[1], matches[2]];
+    },
+
+    render: function() {
+        const BaseDialog = sdk.getComponent('views.dialogs.BaseDialog');
+        const Loader = sdk.getComponent("elements.Spinner");
```

Comment:
Quotes should be consistent. I prefer `Spinner` over `Loader` but we use both throughout so :man_shrugging:.